Cesium3DTilesSelection/src/CesiumIonTilesetLoader.cpp

```diff
@@ -408,53 +518,88 @@ TileChildrenResult CesiumIonTilesetLoader::createTileChildren(
   return pLoader->createTileChildren(tile, ellipsoid);
 }
 
-void CesiumIonTilesetLoader::refreshTokenInMainThread(
-    const std::shared_ptr<spdlog::logger>& pLogger,
-    const std::shared_ptr<CesiumAsync::IAssetAccessor>& pAssetAccessor,
-    const CesiumAsync::AsyncSystem& asyncSystem) {
-  if (this->_refreshTokenState == TokenRefreshState::Loading) {
-    return;
+CesiumAsync::SharedFuture<std::string>
+CesiumIonTilesetLoader::refreshTokenInMainThread(
```

Comment:
There shouldn't be any blocking of the main thread, it's all Future based. The "in main thread" part of the name is meant to express the requirement that the method only be called from the main thread.

Reply:
> There shouldn't be any blocking of the main thread, it's all Future based. The "in main thread" part of the name is meant to express the requirement that the method only be called from the main thread.

Doesn't that mean that `this->_tokenRefreshInProgress->wait()` is called in the main thread?

Reply:
Oh yes, that's true, but we only call `wait` if `isReady()` returns true. `wait` is guaranteed not to block when the Future is already ready. This is documented behavior on the `isReady` method. It will return the value or throw an exception immediately.